node-tests/blueprints/component-test-test.js

```diff
@@ -38,6 +38,14 @@ describe('Blueprint: component-test', function () {
           );
         });
       });
+
+      it('component-test x-foo --strict', function () {
+        return emberGenerateDestroy(['component-test', 'x-foo', '--strict'], (_file) => {
+          expect(_file('tests/integration/components/x-foo-test.gjs')).to.equal(
+            fixture('component-test/rfc232.gjs')
```

Comment:
This fails when running the TypeScript blueprint tests:

```
  1) Blueprint: component-test
       in app
         with default setup
           component-test x-foo --strict:
     AssertionError: expected "tests/integration/components/x-foo-test.gjs" to exist

parent path "tests/integration/components" exists and contains:
- x-foo-test.gts
      at /.../ember.js/node-tests/blueprints/component-test-test.js:44:75
      at /.../ember.js/node_modules/.pnpm/ember-cli-blueprint-test-helpers@0.19.2/node_modules/ember-cli-blueprint-test-helpers/lib/ember-generate-destroy.js:34:17
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

I'm not sure what the next step is here. The other tests don't check on TypeScript-specific files. Is the expect assumed to automatically work for TS too, but it doesn't here?

Welcoming any input 😄

Reply:
Validating the path dynamically would work:

```js
const filePath = fs.existsSync('tests/integration/components/x-foo-test.gts')
  ? 'tests/integration/components/x-foo-test.gts'
  : 'tests/integration/components/x-foo-test.gjs';
```

But that seems like logic which shouldn't be in the test.

Reply:
You're getting TypeScript output when the TypeScript blueprints should be converted to JS by the blueprint system (unless it is a typescript project). 

I believe this https://github.com/ember-polyfills/ember-cli-typescript-blueprint-polyfill/blob/main/src/utils.js#L13-L15 needs to include gts, but there may be other changes needed for the blueprint system to convert gts to gjs.

Reply:
I can try to find some time to push this over the finish line as well :+1:

Reply:
That would be much appreciated. Realistically, I won't have time to look into this in the near future :/

Reply:
Ok, that was confusing.. :smile: 

I [updated the polyfill](https://github.com/ember-polyfills/ember-cli-typescript-blueprint-polyfill/pull/5) and used the branch here, but that still didn't work because the polyfill [disables itself](https://github.com/ember-polyfills/ember-cli-typescript-blueprint-polyfill/blob/8bede4c079219d3f7a68fedabed08296c89460a0/src/typescript-blueprint-polyfill.js#L11-L13) if it thinks it's no longer needed. It seems we are depending on [ember-cli 4.10](https://github.com/emberjs/ember.js/blob/0f9b3fcebb9d813f438a946f4b7d253e11311672/package.json#L130) which _does_ support .ts -> .js in the blueprints, but not .gts -> .gjs (which was released in 5.5 I think). Once I bump the version to 5.12 the test no longer seems to fail.

What was extra confusing is that there are no tests that verify the .ts version of the blueprints: https://github.com/emberjs/ember.js/issues/20362